k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -138,10 +160,63 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
 
     (_watch(resourceVersion), Closable.ref(close))
   }
+
+  /**
+   * Convert this Watchable into an [[Activity]]
+   * @param resourceVersion whether or not to send the initial resource
+   *                        version of the watched resource. this is a
+   *                        special case due to errors with the
+   *                        ConfigMap interpreter
+   * @param onEvent function called on each [[Watch]] event
+   * TODO: i wish this returned a [[Activity.State]] so that the caller
+   *       of this function could turn a [[Watch.Error]] into an
+   *       [[Activity.Failed]] if they wanted watch errors to fail this
+   *       [[Activity]], although none of our current code exhibits this
+   *       use-case...
+   *         - eliza, 7/18/2017
+   * @return
+   */
+  def activity[T](convert: G => T, resourceVersion: Boolean = true)
+                  (onEvent: (T, W) => T)
+  : Activity[T] =
+    Activity(Var.async[Activity.State[T]](Activity.Pending) { state =>
+      val closeRef = new AtomicReference[Closable](Closable.nop)
+      val pending = get(retryIndefinitely = true)
+        // if the initial GET failed, then the activity is a failure
+        .onFailure { e => state.update(Activity.Failed(e)) }
```

Comment:
I think we should `log.warn` when this happens, to make it easier to track down configuration errors.